server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1254,7 +1254,8 @@ private boolean deleteDataStoreInternal(StoragePoolVO sPool, boolean forced) {
             // Check if the pool has associated volumes in the volumes table
             // If it does , then you cannot delete the pool
             if (vlms.first() > 0) {
-                throw new CloudRuntimeException("Cannot delete pool " + sPool.getName() + " as there are associated volumes for this pool");
+                logger.debug("Cannot delete storage pool {} as the following non-destroyed volumes are on it: {}.", sPool.getName(), getStoragePoolNonDestroyedVolumesLog(sPool.getId()));
```

Comment:
```suggestion
                logger.debug("Cannot delete storage pool {} as the following non-destroyed volumes are on it: {}.", sPool.getName(), () -> getStoragePoolNonDestroyedVolumesLog(sPool.getId()));
```

When executing heavy processes to load data only to log it, call the methods inside a lambda; this way, they will be evaluated only at the right log level.